src/components/forms/CharacterCount/CharacterCount.tsx

```diff
@@ -65,7 +65,7 @@ export const CharacterCount = ({
   getMessage = defaultMessage,
   ...remainingProps
 }: TextInputCharacterCountProps | TextareaCharacterCountProps): JSX.Element => {
-  const initialCount = getCharacterCount(value || defaultValue)
+  const [initialCount] = useState(getCharacterCount(value || defaultValue))
```

Comment:
I think this is fine, so not blocking, but out of curiosity: Why not use a ref here?

Something like

```suggestion
  const initialCount = useRef(null)
  if (intialCount.current === null) {
    intialCount.current = getCharacterCount(value || defaultValue)
  }
```

Admittedly, it looks way grosser, but is supposedly more semantically correct. [Found some relevant discussion/examples here](https://dev.to/raicuparta/compute-values-on-component-mount-with-react-hooks-state-vs-ref-4epk)

Reply:
@brandonlenz - Argh, neither of our approaches is correct here. I'm glad you left a comment.

My approach failed at the one improvement it was supposed to achieve: only evaluate `getCharacterCount(value || defaultValue)` once for the purpose of setting `initialCount` during the initial render. Use that value for the next few `useState()` lines and then forget about it. Because I passed `getCharacterCount(value || defaultValue)` as a value to `useState()` rather than through an initializer function, it _still_ gets evaluated every render. My change only achieved adding tiny overhead to each render cycle.

Your approach does better at only evaluating `getCharacterCount(value || defaultValue)` once like intended (kudos for spotting that and not assigning the value as the `useRef()` argument). Unfortunately, it violates another principle: ref values are not supposed to be accessed during renders. Enabling react-hooks recommended eslint ruleset notes this:

> Error: Cannot access refs during render
> 
> React refs are values that are not needed for rendering. Refs should only be accessed outside of render, such as in event handlers or effects. Accessing a ref value (the `current` property) during render can cause your component not to update as expected (https://react.dev/reference/react/useRef)

In 9d33668, I switch to using initializer functions for this `setState` and also for the message `setState`.